h2o-automl/src/main/java/ai/h2o/automl/modeling/XGBoostSteps.java

```diff
@@ -172,6 +201,108 @@ XGBoostParameters prepareModelParameters() {
+                    XGBoostParameters xgBoostParameters = (XGBoostParameters) bestXGB._parms.clone();
+                    XGBoostParameters defaults = new XGBoostParameters();
+                    xgBoostParameters._ntrees = 10000; // reset ntrees (we'll need more for this fine tuning)
+                    setStoppingCriteria(xgBoostParameters, defaults, SeedPolicy.None); // keep the same seed as the bestXGB
+                    // reset _eta to defaults, otherwise it ignores the _learn_rate hyperparam: this is very annoying!
+                    xgBoostParameters._eta = defaults._eta;
+//                    xgBoostParameters._learn_rate = defaults._learn_rate;
+/*
+                    //keep score_tree_interval fixed, but increase stopping_rounds when lowering learn rate
+                    int sr = xgBoostParameters._stopping_rounds;
+                    Object[][] hyperParams = new Object[][] {
+                            new Object[] {"_learn_rate", "_stopping_rounds"},
+                            new Object[] {        0.5  ,                sr },
+                            new Object[] {        0.2  ,                sr },
+                            new Object[] {        0.1  ,              2*sr },
+                            new Object[] {        0.05 ,              2*sr },
+                            new Object[] {        0.02 ,              3*sr },
+                            new Object[] {        0.01 ,              3*sr },
+                            new Object[] {        0.005,              4*sr },
+                            new Object[] {        0.002,              4*sr },
+                            new Object[] {        0.001,              5*sr },
+                    };
+*/
+                    // keep stopping_rounds fixed, but increases score_tree_interval when lowering learn rate
+                    int sti = xgBoostParameters._score_tree_interval;
+                    Object[][] hyperParams = new Object[][] {
+                            new Object[] {"_learn_rate", "_score_tree_interval"},
+                            new Object[] {       0.5   ,                   sti },
+                            new Object[] {       0.2   ,                   sti },
+                            new Object[] {       0.1   ,                 2*sti },
+                            new Object[] {       0.05  ,                 2*sti },
+                            new Object[] {       0.02  ,                 3*sti },
+                            new Object[] {       0.01  ,                 3*sti },
+                            new Object[] {       0.005 ,                 4*sti },
+                            new Object[] {       0.002 ,                 4*sti },
+                            new Object[] {       0.001 ,                 5*sti },
+                            new Object[] {       0.0005,                 5*sti },
+                    };
```

Comment:
I tried 2 strategies to keep the model training speed reasonnable when the learn_rate is diminishing:
1. increasing `stopping_rounds` when `learn_rate` is lowered, as first suggested in the Jira ticket: we keep scoring regulary (every 5 trees: this is hardcoded in XGBoostSteps as this can not be specified by the user), this way the training doesn't stop "too early" with low learn_rate, but it probably scores too often. Also, we modify a parameter specified by the user, which is not polite :)
2. keep the `stopping_rounds` constant and as specified by the user, but instead increase the `score_tree_interval` when `learn_rate`is lowered: this allows to score less often, but then we have more trees being built between each score, and for each `stopping_round`, so this also avoids stopping "too early" with low `learn_rate`.

Reply:
@sebhrusen Second approach sounds more reasonable, computationally.